azurelinuxagent/ga/utils/exthandler_utils.py

```diff
@@ -0,0 +1,343 @@
+                "field - {0}, {1}/ value - {2}, {3}".format("cores",
+                                                            "limit_percentage",
+                                                            cores,
+                                                            limit_percentage))
+        self.cores = int(cores)
+        self.limit_percentage = float(limit_percentage)
+
+    def __eq__(self, other):
+        return self.cores == other.cores
+
+    def __lt__(self, other):
+        return self.cores < other.cores
+
+    def __gt__(self, other):
+        return self.cores > other.cores
+
+    def __str__(self):
+        return {"cores": self.cores, "limit_percentage": self.limit_percentage}.__str__()
+
+    def __repr__(self):
+        return {"cores": self.cores, "limit_percentage": self.limit_percentage}.__str__()
+
+
+class MemoryLimits(object):
+    def __init__(self, memory_node):
+        if "max_limit_MBs" in memory_node and "max_limit_percentage" in memory_node:
+            self.max_limit_percentage = memory_node.get("max_limit_percentage", None)
+            self.max_limit_MBs = memory_node.get("max_limit_MBs", None)
+
+            if type(self.max_limit_percentage) is not float and type(self.max_limit_percentage) is not int:
+                raise ExtensionHandlerConfigurationError(
+                    "Incorrect types for Memory values - field - {0} | value - {1} | type - {2}".format(
+                        "max_limit_percentage",
+                        self.max_limit_percentage,
+                        type(self.max_limit_percentage)))
+
+            if type(self.max_limit_MBs) is not float and type(self.max_limit_MBs) is not int:
+                raise ExtensionHandlerConfigurationError(
+                    "Incorrect types for Memory values - field - {0} | value - {1} | type - {2}".format(
+                        "max_limit_MBs", self.max_limit_MBs, type(self.max_limit_MBs)))
+
+            if self.max_limit_percentage > 100 or self.max_limit_percentage < 1:
+                raise ExtensionHandlerConfigurationError(
+                    "Incorrect types for Memory values - field - {0} | value - {1} | type - {2}".format(
+                        "max_limit_MBs", self.max_limit_MBs, type(self.max_limit_MBs)))
+        else:
+            raise ExtensionHandlerConfigurationError(
+                "Default max memory limit not set. max_limit_MBs: {0}, max_limit_percentage: {1}".format(
+                    memory_node.get("max_limit_MBs", None), memory_node.get("max_limit_percentage", None)))
+
+        self.memory_pressure_warning = memory_node.get("memory_pressure_warning", None)
+        self.memory_oom_kill = None  # default will be set in compute_default for memory flags.
+
+        if "memory_oom_kill" in memory_node:
+            if memory_node["memory_oom_kill"].lower() in MEMORY_OOM_KILL_OPTIONS:
+                self.memory_oom_kill = memory_node["memory_oom_kill"].lower()
```

Comment:
oh, so sometimes we are case-insensitive 

we should be consistent

Reply:
We aren't being case-insensitive about the key, but the value, in this case. Keys are strict.

Reply:
why the difference?

Reply:
the keys need to be strict - as we would be documenting them and don't want to add ambiguity in the key-names. The values can be made strict (enumify) and well documented. I can add that part of validation later.